smallrye-reactive-messaging-amqp/src/test/java/io/smallrye/reactive/messaging/amqp/AmqpSinkTest.java

```diff
@@ -242,12 +253,77 @@ public void testSinkUsingObject() throws Exception {
         assertThat(count.get()).isEqualTo(msgCount);
     }
 
+    /**
+     * Extension of Person that cannot be serialized to JSON.
+     */
+    public static class Bad extends Person {
+
+        public Person getPerson() {
+            return this;
+        }
+
+    }
+
+    @Test
+    @Timeout(30)
+    public void testSinkUsingObjectThatCannotBeSerialized() throws Exception {
+        int msgCount = 10;
+        AtomicInteger nack = new AtomicInteger();
+        CountDownLatch msgsReceived = new CountDownLatch(msgCount / 2);
+        List<org.apache.qpid.proton.message.Message> messagesReceived = Collections
+                .synchronizedList(new ArrayList<>(msgCount));
+
+        server = setupMockServerForTypeTest(messagesReceived, msgsReceived);
+
+        SubscriberBuilder<? extends Message<?>, Void> sink = createProviderAndSink(UUID.randomUUID().toString(),
+                server.actualPort());
+        //noinspection unchecked
+        Multi.createFrom().range(0, 10)
+                .map(i -> {
+                    Person p;
+                    if (i % 2 == 0) {
+                        p = new Person();
+                        p.setName(HELLO + i);
+                    } else {
+                        p = new Bad();
+                        p.setName(HELLO + i);
+                    }
+                    return p;
+                })
+                .map(p -> Message.of(p, () -> CompletableFuture.completedFuture(null), e -> {
+                    nack.incrementAndGet();
+                    return CompletableFuture.completedFuture(null);
+                }))
+                .subscribe((Subscriber<? super Message<Person>>) sink.build());
+
+        assertThat(msgsReceived.await(6, TimeUnit.SECONDS)).isTrue();
+        await().until(() -> nack.get() == 5);
+
+        AtomicInteger count = new AtomicInteger();
+        messagesReceived.forEach(msg -> {
+            assertThat(msg.getContentType()).isEqualTo("application/json");
+
+            Section body = msg.getBody();
+            assertThat(body).isInstanceOf(Data.class);
+            Binary bin = ((Data) body).getValue();
+            byte[] bytes = Binary.copy(bin).getArray();
+            JsonObject json = Buffer.buffer(bytes).toJsonObject();
+            Person p = json.mapTo(Person.class);
+            assertThat(p.getName()).startsWith("hello-");
+            count.incrementAndGet();
+        });
+
+        assertThat(count.get()).isEqualTo(msgCount / 2);
+        assertThat(nack).hasValue(5);
```

Comment:
This is useless, because we already do `await().until(() -> nack.get() == 5)` above, but not a big deal :-)

Reply:
Ah ah, I added the await after this line (because obviously, it was racy).